client/src/Components/Forms/Login/index.tsx

```diff
@@ -25,12 +22,50 @@ const LoginSchema = Yup.object().shape({
     .required('Required')
 })
 
-export const LoginForm: React.FC<LoginFormProps> = ({ onSubmit }) => {
+export const LoginForm: React.FC = () => {
+  const [error, setError] = useState('')
+
+  const authContext: IAuthContext = useContext(AuthContext)
+  const { login } = authContext
+
+  const history = useHistory()
+
+  const handleSubmit = async (values: LoginFormValues) => {
+    setError('')
+    try {
+      await login(
+        // Firebase requires a 'email' username and passwords to be at least 6 characters
+        `${values.username}@FBSF.com`,
+        `${values.year.toString()}FBSF`
+      )
+      history.push('/')
+    } catch (error) {
+      console.log(error)
+      if (error.code === 'auth/user-not-found') {
+        setError('Username cannot be found')
+      } else if (error.code === 'auth/wrong-password') {
+        setError('Password is incorrect')
+      } else if (error.code === 'auth/too-many-requests') {
+        setError('Too many login attempts, try again later')
+      } else setError('Error while logging in')
+    }
+  }
+
+  // const handleLogout = async () => {
+  //   setError('')
+  //   try {
+  //     await logout()
+  //     history.push('/login')
+  //   } catch {
+  //     setError('Error while logging out')
+  //   }
+  // }
```

Comment:
![Delet this](https://www.meme-arsenal.com/memes/2b123f9b2646706892fa0286a1425bc9.jpg)

Reply:
hahahahahahahahaha